neuron_morphology/transforms/tilt_correction/compute_tilt_correction.py

```diff
@@ -0,0 +1,226 @@
+from typing import Dict, Optional, List, Any
+
+import logging
+import copy as cp
+
+import numpy as np
+from scipy.spatial.distance import euclidean
+import pandas as pd
+import h5py
+
+from argschema.argschema_parser import ArgSchemaParser
+from neuron_morphology.transforms.tilt_correction._schemas import (
+    InputParameters, OutputParameters)
+from neuron_morphology.morphology import Morphology
+from neuron_morphology.swc_io import morphology_from_swc
+import neuron_morphology.transforms.affine_transform as aff
+
+
+CCF_SHAPE = (1320, 800, 1140)
+CCF_RESOLUTION = 10
+
+
+def get_tilt_correction(morphology: Morphology,
+                        soma_voxel: Dict[str, int],
+                        slice_angle_matrix: float,
+                        closest_path,
+                        ):
+    """
+        Find the tilt angle between the slice plane and the nearest streamline
+
+        Parameters
+        ----------
+        morphology: Morphology object
+        soma_voxel: soma voxel in ccf {'x': , 'y': , 'z': }
+        slice_angle_matrix: 4 x 4 affine matrix of the slice plane
+            relative to ccf
+        closest_path: 3 x N array of voxel coordinates in closest streamline
+
+        Returns
+        -------
+        tilt angle correction (radians)
+
+    """
+
+    # Find slice plane vector
+    M = slice_angle_matrix[0:3, :]
+    a2 = np.dot(M, np.array([0, 0, 0, 1]))
+    b2 = np.dot(M, np.array([1, 0, 0, 1]))
+    c2 = np.dot(M, np.array([0, 1, 0, 1]))
+
+    norm_vec = np.cross(b2 - a2, c2 - a2)
+    norm_unit = norm_vec / euclidean(norm_vec, [0, 0, 0])
+
+    # Find approximate streamline vector
+    if euclidean(closest_path[:, 0], soma_voxel) == 0:
+        # if soma is at the top, calculate from other end of streamline
+        streamline_unit = (soma_voxel - closest_path[:, -1]) /\
+            euclidean(soma_voxel, closest_path[:, -1])
+    else:
+        streamline_unit = (closest_path[:, 0] - soma_voxel) /\
+             euclidean(closest_path[:, 0], soma_voxel)
+
+    # determine angle between
+    s = np.linalg.norm(np.cross(norm_unit, streamline_unit))
+    dp = np.dot(norm_unit, streamline_unit)
+    theta = np.arctan2(s, dp)
+
+    tilt_angle = np.pi / 2 - theta
```

Comment:
theta is between -Pi and Pi. This means that tilt_angle can end up being negative. Have you considered this scenario?

Reply:
Yes negative rotations are valid. In fact with the flip toggle, they seem semi likely.

Reply:
Let's say you get theta=75 degrees, then tilt=15. 
But if the normal was facing in the opposite direction than you would get theta=180-75 =105 and tilt=90-105=-15. So, direction of the normal can cause tilt to be applied in the wrong direction.